tools/souper-check.cpp

```diff
@@ -96,6 +121,93 @@ int SolveInst(const MemoryBufferRef &MB, Solver *S) {
   int Ret = 0;
   int Success = 0, Fail = 0, Error = 0;
   for (auto Rep : Reps) {
+    if (InferNeg) {
+      bool Negative;
+      if (std::error_code EC = S->negative(Rep.BPCs, Rep.PCs, Rep.Mapping.LHS,
+                                           Negative, IC)) {
+        llvm::errs() << EC.message() << '\n';
+      }
+      std::string s;
+      if (Negative)
+        s = "true";
+      else
+        s = "false";
```

Comment:
avoid this repeated code for making a string out of a bool. either factor it out into a helper function or else use the native C++ method for printing a bool as a string.

Reply:
std::boolalpha didn't work in context of llvm::outs(), so I added a small helper function.